### PR TITLE
Add `prefer-node-append` rule

### LINKED ISSUE #170: Rule proposal: `prefer-node-append`
Prefer `node.append()` over `node.appendChild()`.

[`.append()`](https://developer.mozilla.org/en-US/docs/Web/API/ParentNode/append) has many benefits over `.appendChild()`.

### PR DESCRIPTION
**Pull request by nikhilsaldanha**
- Created a new entry in index.js for prefer-node-append.

 - Added a test for prefer-node-append.

closes #170

### CODE REVIEW

File: test/prefer-node-append.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+				message: 'Prefer `append` over `appendChild`'
+			}]
+		},
+		{
+			code: 'document.body.appendChild(child);',
+			output: 'document.body.append(child);',
+			errors: [{
+				message: 'Prefer `append` over `appendChild`'
+			}]
+		}
```
Comment (MrHen) [inline test/prefer-node-append.js]: Add tests for `appendChild()` and `appendChild(null)` just to make sure eslint doesn't crash.

**Comment by MrHen** [comment]
Need to add the rule to readme.md
https://github.com/sindresorhus/eslint-plugin-unicorn#usage

File: rules/prefer-node-append.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+const getDocsUrl = require('./utils/get-docs-url');
+
+const getMethodName = memberExpression => memberExpression.property.name;
+
+const fix = (fixer, sourceCode, callNode, memberExpression) => {
+	const nodeCode = sourceCode.getText(memberExpression.object);
+	const argumentsCode = sourceCode.getText(callNode.arguments[0]);
+	const fixedCodeStatement = `${nodeCode}.append(${argumentsCode})`;
+	return fixer.replaceText(callNode, fixedCodeStatement);
+};
```
Comment (sindresorhus) [inline rules/prefer-node-append.js]: This could be simplified a lot. Don't get the source code and construct it manually. Just replace the property at the node level.

Something like:

```js
fixer.replaceText(memberExpression.property, 'append');
```

**Comment by sindresorhus** [comment]
Look at some previous additions to see what's required for a new rule:

- https://github.com/sindresorhus/eslint-plugin-unicorn/commit/5dd529f2e1cfa84652e880a079620640286c5aea
- https://github.com/sindresorhus/eslint-plugin-unicorn/commit/9f7f811760b34df99d7dfd5d814904121740895c

**Comment by nikhilsaldanha** [comment]
All changes requested for are done

**Comment by sindresorhus** [comment]
Nice work, @nikhilsaldanha :)
